lib/ansible/executor/task_executor.py

```diff
@@ -399,6 +396,33 @@ def _run_loop(self, items):
 
         return results
 
+    def _get_delegated_vars(self, templar, variables):
+        if self._task.delegate_to:
+            delegated_host_name = templar.template(self._task.delegate_to, fail_on_undefined=False)
+            delegated_host = self._variable_manager._inventory.get_host(delegated_host_name)
+            if delegated_host is None:
+                for h in self._variable_manager._inventory.get_hosts(ignore_limits=True, ignore_restrictions=True):
+                    # check if the address matches, or if both the delegated_to host
+                    # and the current host are in the list of localhost aliases
+                    if h.address == delegated_host_name:
+                        delegated_host = h
+                        break
+                else:
+                    delegated_host = Host(name=delegated_host_name)
+
+            variables['ansible_delegated_vars'] = {
```

Comment:
agreed, but we already have too many memory baloons and we are adding/overriding a specific key that would have to be wiped out anyways in case it had old/prev loop/task data.